### DISCUSSION TITLE
Is there a general method to automatically capture the startup port when the Java service is started?

### DISCUSSION THREAD

**Discussion by zhangjiabin1010** [open-telemetry/opentelemetry-java-instrumentation#11918] [Q&A] [answered]
Without the need to manually configure environment variables。How can I achieve this functionality

**Comment by laurit** [comment]
What do you mean by `startup port`?

**Reply by zhangjiabin1010** [reply]
> 你是什​​么意思`startup port`？

The server.port specified when starting the service.

**Reply by laurit** [reply, marked as answer]
I don't think we have anything for this.

**Comment by jaydeluca** [comment]
Could you provide more details into what you are trying to do and why?

> Without the need to manually configure environment variables

Is there an environment variable you are currently setting for this? Which one(s)?
